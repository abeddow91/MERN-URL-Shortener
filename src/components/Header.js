import React from 'react';

const Header = ({ message }) => {
  return (
    <h2 className="Header text-center">
    {message}
    </h2>
  );
};

Header.propTypes = {
  message: React.PropTypes.string.isRequired
};

Header.defaultProps = {
  message: 'This is the default header message'
};


export default Header;
