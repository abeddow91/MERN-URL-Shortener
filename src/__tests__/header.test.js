import React from 'react';
import ReactDOM from 'react-dom';
import ReactTestUtils from 'react-addons-test-utils';
import {shallow, mount} from 'enzyme';
import toJson from 'enzyme-to-json';

import Header from '../components/Header';

//This test mounts a component and checks that it doesn't throw an exception during rendering.

    it('renders without crashing', () => {
      mount(<Header />);
    });

  it('should render the Header', () => {
    const header = shallow(<Header />);
    const tree = toJson(header);
    expect(tree).toMatchSnapshot()
  });
