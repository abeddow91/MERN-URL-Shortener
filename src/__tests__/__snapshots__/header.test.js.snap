// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Header 1`] = `
<h1
  className="Header text-center"
>
  Url Shortener
</h1>
`;
